File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/net/packet/CBPacketRemoveEntity.java
package pize.tests.voxelgame.main.net.packet;

import pize.net.tcp.packet.IPacket;
import pize.net.tcp.packet.PacketHandler;
import pize.util.io.PizeInputStream;
import pize.util.io.PizeOutputStream;
import pize.tests.voxelgame.main.entity.Entity;

import java.io.IOException;
import java.util.UUID;

public class CBPacketRemoveEntity extends IPacket<PacketHandler>{
    
    public static final int PACKET_ID = 16;
    
    public CBPacketRemoveEntity(){
        super(PACKET_ID);
    }
    
    
    public UUID uuid;
    
    public CBPacketRemoveEntity(Entity entity){
        this();
        uuid = entity.getUUID();
    }
    
    
    @Override
    protected void write(PizeOutputStream stream) throws IOException{
        stream.writeUTF(uuid.toString());
    }
    
    @Override
    public void read(PizeInputStream stream) throws IOException{
        uuid = UUID.fromString(stream.readUTF());
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/net/packet/CBPacketRemoveEntity.kt
package pize.tests.voxelgame.main.net.packet

import pize.net.tcp.packet.IPacket
import pize.net.tcp.packet.PacketHandler
import pize.tests.voxelgame.main.entity.Entity
import pize.util.io.PizeInputStream
import pize.util.io.PizeOutputStream
import java.io.IOException
import java.util.*

class CBPacketRemoveEntity() : IPacket<PacketHandler?>(PACKET_ID) {
    var uuid: UUID? = null

    constructor(entity: Entity?) : this() {
        uuid = entity.getUUID()
    }

    @Throws(IOException::class)
    override fun write(stream: PizeOutputStream?) {
        stream!!.writeUTF(uuid.toString())
    }

    @Throws(IOException::class)
    override fun read(stream: PizeInputStream?) {
        uuid = UUID.fromString(stream!!.readUTF())
    }

    companion object {
        const val PACKET_ID = 16
    }
}